src/MultiPartStreamFactory.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace Http\Message;
+
+use Psr\Http\Message\StreamInterface;
+
+/**
+ * Factory for creating Multipart Streams.
+ *
+ * @author Robert Schönthal <robert.schoenthal@gmail.com>
+ */
+interface MultiPartStreamFactory extends StreamFactory
```

Comment:
Would a class use both streams and multipart streams? Or is that just a edge case?

If a class are likely to use both, then yes, the should extend.